python/ray/util/sgd/v2/backends/backend.py

```diff
@@ -121,18 +143,84 @@ def _load_checkpoint(self,
 
     def write_checkpoint(self, checkpoint: Dict):
         """Writes checkpoint to disk."""
-        if self._checkpoint_strategy.num_to_keep == 0:
+        num_to_keep = self._checkpoint_strategy.num_to_keep
+
+        if num_to_keep == 0:
             # Checkpoints should not be persisted to disk.
             return
 
-        # TODO(matt): Implement additional checkpoint strategy functionality.
-        # Get or create checkpoint dir.
-        self.latest_checkpoint_dir.mkdir(parents=True, exist_ok=True)
-        # Write checkpoint to disk.
-        with self.latest_checkpoint_path.open("wb") as f:
-            cloudpickle.dump(checkpoint, f)
-            logger.debug(f"Checkpoint successfully written to: "
-                         f"{self.latest_checkpoint_path}")
+        checkpoint_score_attribute = \
+            self._checkpoint_strategy.checkpoint_score_attribute
+        checkpoint_score_order = \
+            self._checkpoint_strategy.checkpoint_score_order
+        if checkpoint_score_attribute not in checkpoint:
+            raise ValueError(f"Unable to persist checkpoint for "
+                             f"checkpoint_score_attribute: "
+                             f"{checkpoint_score_attribute}. "
+                             f"Include this attribute in the call to "
+                             f"save_checkpoint.")
+        checkpoint_score = checkpoint[checkpoint_score_attribute]
+        try:
+            checkpoint_score = float(checkpoint_score)
```

Comment:
You might want to use `numbers.Numbers` to check instead of trying to cast to float.

https://stackoverflow.com/questions/29773143/how-to-determine-that-a-python-object-is-numeric